dvd/dvd_dump.ixx

```diff
@@ -735,6 +937,14 @@ export bool redumper_dump_dvd(Context &ctx, const Options &options, DumpMode dum
             s += sectors_to_read;
     }
 
+    if(is_xbox)
+    {
+        // re-unlock drive before returning
+        status = cmd_kreon_set_lock_state(*ctx.sptd, KREON_LockState::WXRIPPER);
+        if(status.status_code)
+            LOG("warning: failed to unlock drive at end of dump, SCSI ({})", SPTD::StatusMessage(status));
+    }
```

Comment:
This last unlock is not explicitly required, however the thought was to put the drive in the unlocked state as that appears to be the "default" state. The starting unlock is still needed as if another program is not "kind" and leaves the drive locked then we cannot detect if there is an XGD present. By unlocking the drive we also make sure we are not an "un-kind" program to any other programs that may use similar detection methods.

Reply:
None that I am aware of. I have not seen any other tool or the documentation mention such a method

Reply:
Certainly no documented Kreon commands to get the lock state.
Fortunately from our testing, running unlock when it is already unlocked, or if a non-xbox disc is present, seems to do nothing.

Reply:
Got it.